src/gui/MainWindow.cpp

```diff
@@ -644,17 +649,31 @@ void MainWindow::resetWindowTitle()
 bool MainWindow::mayChangeProject(bool stopPlayback)
 {
 	if( stopPlayback )
+	{
 		Engine::getSong()->stop();
+	}
 
-	if( !Engine::getSong()->isModified() )
+	if( !Engine::getSong()->isModified() && !isRecoverSession() )
 	{
 		return( true );
 	}
 
-	QMessageBox mb( tr( "Project not saved" ),
-				tr( "The current project was modified since "
+	// Separate message strings for modified and recovered files
+	QString messageTitleRecovered = tr( "Recovered project not saved" );
+	QString messageRecovered = tr( "This project was recovered from the "
+					"previous session. It is currently "
+					"unsaved and will be lost if you don't "
+					"Save it. Do you want to save it now?" );
+
+	QString messageTitleUnsaved = tr( "Project not saved" );
```

Comment:
Since the strings are translated at creation time by using tr() it should be Ok. You can then copy to an other variable because the content is already translated.
So this should work as expected :)